govcd/org_test.go

```diff
@@ -16,7 +16,7 @@ import (
 func (vcd *TestVCD) Test_RefreshOrg(check *C) {
 
 	if vcd.skipAdminTests {
```

Comment:
It's not visible in this change set alone, but we have at least three code paths that determine whether configuration uses sys admin.

`api_vcd_test.go` has:
```
	vcd.client = vcdClient
	if config.Provider.SysOrg != "System" {
		fmt.Printf("Skipping OrgAdmin tests\n")
		vcd.skipAdminTests = true
	}
```

`api_vcd.go` has:
```
vdcCli.Client.IsSysAdmin = false
	if "System" == org {
		vdcCli.Client.IsSysAdmin = true
	}
```

And Terraform provider `vcd/config_test.go` has:
```
func usingSysAdmin() bool {
		sysOrg := strings.ToLower(testConfig.Provider.SysOrg)
		org := strings.ToLower(testConfig.VCD.Org)
		if sysOrg == "system" || (sysOrg == "" && org == "system") {
			return true
		}
		return false
	}
```

A bit refactoring needed to collapse these code paths into a single one.

Reply:
The code in `api_vcd.go` was created after the test skipping in `api_vcd_test.go` with a different purpose that is not related to test. Given thats, we can reuse `vcd.Client.isSysAdmin` for test skipping.

Terraform tests and go-vcloud-director tests are not related and could not reuse this kind of code. However, given that we now have a `isSysAdmin` property in the connector (again, it was added later) we can change the Terraform code to depend on that.

Reply:
done